crates/markdown_preview/src/markdown_preview_view.rs

```diff
@@ -0,0 +1,166 @@
+use editor::{Editor, EditorEvent};
+use gpui::{
+    div, list, AnyElement, AppContext, EventEmitter, FocusHandle, FocusableView,
+    InteractiveElement, IntoElement, ListState, ParentElement, Render, Styled, View, ViewContext,
+};
+use language::LanguageRegistry;
+use std::sync::Arc;
+use ui::prelude::*;
+use workspace::item::Item;
+use workspace::Workspace;
+
+use crate::OpenPreview;
+
+pub struct MarkdownPreviewView {
+    focus_handle: FocusHandle,
+    list_state: ListState,
+}
+
+impl MarkdownPreviewView {
+    pub fn register(workspace: &mut Workspace, _cx: &mut ViewContext<Workspace>) {
+        let languages = workspace.app_state().languages.clone();
+
+        workspace.register_action(move |workspace, _: &OpenPreview, cx| {
+            if workspace.has_active_modal(cx) {
+                cx.propagate();
+                return;
+            }
+            let languages = languages.clone();
+            let editor = workspace.active_item_as::<Editor>(cx).unwrap();
+            MarkdownPreviewView::deploy_preview(workspace, languages, editor, cx);
+            cx.notify();
+        });
+    }
+
+    pub fn new(
+        active_editor: View<Editor>,
+        languages: Arc<LanguageRegistry>,
+        cx: &mut ViewContext<Self>,
+    ) -> Self {
+        let focus_handle = cx.focus_handle();
+
+        cx.subscribe(&active_editor, |_this, _editor, event: &EditorEvent, cx| {
+            if *event == EditorEvent::Edited {
+                cx.notify();
+            }
+        })
+        .detach();
+
+        let list_state = ListState::new(1, gpui::ListAlignment::Top, px(1000.), move |_ix, cx| {
```

Comment:
I'll see if I can push a commit that removes this intermediary list :)

Reply:
I got your all's suggestions working in my new PR: https://github.com/zed-industries/zed/pull/6958! Thanks for the help on that. I'll apply the rest of these comments on that PR as well.